libs/judicial-system/court-client/src/lib/courtClient.service.ts

```diff
@@ -76,24 +80,48 @@ export class CourtClientService {
       await this.login(clientId)
     }
 
-    try {
-      const res = await request(this.authenticationToken[clientId])
-
-      return stripResult(res)
-    } catch (error) {
-      if (isRetry) {
-        this.logger.error('Error while calling the court service', error)
-
-        throw error
-      }
-
-      this.logger.warn(
-        'Error while calling the court service - attempting relogin',
-        error,
-      )
-
-      return this.wrappedRequest(clientId, request, true)
-    }
+    return request(this.authenticationToken[clientId])
+      .then((res) => stripResult(res))
+      .catch((reason) => {
+        const { statusCode, body } = reason as {
+          statusCode: number
+          body: string
+        }
+
+        if (
+          !isRetry &&
+          statusCode === 400 &&
+          body?.startsWith('authenticationToken is expired')
```

Comment:
Making decisions based on whether a hard coded string is in the response body of a request seems fragile to me. Are we sure there is no better way to check for this scenario?